quantum/rgblight.c

```diff
@@ -677,33 +683,55 @@ void rgblight_sethsv_slave(uint16_t hue, uint8_t sat, uint8_t val) {
 
 #ifndef RGBLIGHT_CUSTOM_DRIVER
 void rgblight_set(void) {
-  LED_TYPE *start_led = led + clipping_start_pos;
+  #ifdef RGBLIGHT_LED_MAP
+    LED_TYPE led_buffer[RGBLED_NUM];
+    LED_TYPE *led_dst = led_buffer;
+  #else
+    LED_TYPE *led_dst = led;
+  #endif
+  LED_TYPE *start_led = led_dst + clipping_start_pos;
   uint16_t num_leds = clipping_num_leds;
+
   if (rgblight_config.enable) {
-    #ifdef RGBLIGHT_LED_MAP
-      LED_TYPE led0[RGBLED_NUM];
-      for(uint8_t i = 0; i < RGBLED_NUM; i++) {
-          led0[i] = led[pgm_read_byte(&led_map[i])];
-      }
-      start_led = led0 + clipping_start_pos;
-    #endif
-    #ifdef RGBW
-      ws2812_setleds_rgbw(start_led, num_leds);
-    #else
-      ws2812_setleds(start_led, num_leds);
-    #endif
+    for (uint8_t i = 0; i < RGBLED_NUM; i++) {
+      #ifdef RGBLIGHT_LED_MAP
+        uint8_t led_src_idx = pgm_read_byte(&led_map[i]);
+      #else
+        uint8_t led_src_idx = i;
+      #endif
+
+      #ifdef USE_CIE1931_CURVE
+        led_dst[i].r = pgm_read_byte(&CIE1931_CURVE[led[led_src_idx].r]);
+        led_dst[i].g = pgm_read_byte(&CIE1931_CURVE[led[led_src_idx].g]);
+        led_dst[i].b = pgm_read_byte(&CIE1931_CURVE[led[led_src_idx].b]);
+        #ifdef RGBW
+          led_dst[i].w = pgm_read_byte(&CIE1931_CURVE[led[led_src_idx].w]);
+        #endif
+      #else
+        led_dst[i].r = led[led_src_idx].r;
+        led_dst[i].g = led[led_src_idx].g;
+        led_dst[i].b = led[led_src_idx].b;
+        #ifdef RGBW
+          led_dst[i].w = led[led_src_idx].w;
+        #endif
+      #endif
+    }
   } else {
     for (uint8_t i = 0; i < RGBLED_NUM; i++) {
-      led[i].r = 0;
-      led[i].g = 0;
-      led[i].b = 0;
+      led_dst[i].r = 0;
+      led_dst[i].g = 0;
+      led_dst[i].b = 0;
+      #ifdef RGBW
+        led_dst[i].w = 0;
+      #endif
     }
-    #ifdef RGBW
-      ws2812_setleds_rgbw(start_led, num_leds);
-    #else
-      ws2812_setleds(start_led, num_leds);
-    #endif
   }
+
+  #ifdef RGBW_BB_TWI
+    ledbbtwi_setleds(start_led, num_leds);
```

Comment:
I like this in principle, but I think this may be better as basically some sort of optional function, defined weakly. in rgblight.c.   That way, it isn't necessarily locked to i2c/twi, and some other handling could be added. 

And this way, the ergodox ez specific stuff could be handled there, rather than as a driver. 

I could be overthinking/overcomplicating things there, but ....